scripts/o3de/tests/ExportScripts/test_export_source_built_project.py

```diff
@@ -0,0 +1,1296 @@
+}
+""")
+
+    dummy_commands_file = tmpdir / '.o3de' / '.command_settings'
+    with dummy_commands_file.open('w') as dcf:
+        dcf.write(create_dummy_commands_settings_file(
+            build_config=dummy_build_config, tool_config=dummy_tool_config, archive_format=dummy_archive_format,
+            fail_asset_errors=dum_fail_asset_err, build_assets=dum_build_assets, build_tools=dum_build_tools,
+            build_game_launcher=dum_build_game, build_server_launcher=dum_build_server, build_unified_launcher=dum_build_unified,
+            engine_centric=dum_engine_centric, monolithic=dum_monolithic, allow_reg_overrides=dum_reg_override))
+    
+    with patch('o3de.manifest.get_o3de_folder') as mock_get_o3de_folder,\
+         patch('o3de.export_project.get_default_asset_platform', return_value='pc') as mock_get_asset_platform,\
+         patch('export_source_built_project.export_standalone_project') as mock_export_func:
+        
+        mock_get_o3de_folder.return_value = pathlib.Path(tmpdir.join('.o3de').realpath())
+
+        mock_ctx = create_autospec(O3DEScriptExportContext)
+        mock_ctx.project_path = test_project_path
+        mock_ctx.engine_path = test_engine_path
+        mock_ctx.project_name = test_project_name
+        
+        mock_logger = create_autospec(logging.Logger)
+        test_export_config = exp.get_export_project_config(project_path=None)
+
+        mock_platform='pc'
+        mock_archive_format='none'
+
+        for i in range(5):
+            use_asset_arg = True
+            use_build_tool_arg = True
+            
+            use_asset_fail_arg = True
+            use_mono = True
+            
+            use_engine_centric, use_reg_override = True, True 
+            use_game_build, use_server_build, use_unified_build = True, True, True
+            for product_entry in product(
+                        [True, False],
+                        [True, False]):
+                    
+                    build_config = 'release'
+                    tool_config = 'profile'
+
+                    # batching arguments will help us speed up the testing loop, since we only need to individually verify booleans
+                    if i == 0:
+                        use_build_tool_arg = product_entry[0]
+                        use_asset_arg = product_entry[1]
+                    elif i == 1:
+                        use_asset_fail_arg = product_entry[0]
+                        use_mono = product_entry[1]
+                    elif i == 2:
+                        use_engine_centric = product_entry[0]
+                        use_reg_override = product_entry[1]
+                    elif i == 3:
+                        use_game_build = product_entry[0]
+                        use_server_build = product_entry[1]
+                    elif i == 4:
+                        use_unified_build = product_entry[0]
+                        use_asset_arg = product_entry[1]
```

Comment:
FYI:
python 3.10 now comes with a switch like mechanism: [match](https://peps.python.org/pep-0622/)

```suggestion
                    match i:
                        case 0:
                            use_build_tool_arg = product_entry[0]
                            use_asset_arg = product_entry[1]
                        case 1:
                            use_asset_fail_arg = product_entry[0]
                            use_mono = product_entry[1]
                        case 2:
                            use_engine_centric = product_entry[0]
                            use_reg_override = product_entry[1]
                        case 3:
                            use_game_build = product_entry[0]
                            use_server_build = product_entry[1]
                        case 4:
                            use_unified_build = product_entry[0]
                            use_asset_arg = product_entry[1]
```